projects/dotcms-field-elements/src/components/dot-checkbox/dot-checkbox.tsx

```diff
@@ -1,58 +1,139 @@
 })
 export class DotCheckboxComponent {
+    @Element() el: HTMLElement;
+
+    @Prop() disabled = false;
+    @Prop() name: string;
     @Prop() label: string;
     @Prop() hint: string;
     @Prop() options: string;
-    @Prop() value: string;
-    @Event() onChange: EventEmitter;
+    @Prop() required: boolean;
+    @Prop() requiredmessage: string;
+    @Prop({ mutable: true }) value: string;
+
+    @Event() valueChange: EventEmitter;
+    @Event() statusChange: EventEmitter;
 
     @State() _options: DotOption[];
-    @State() _value: string;
-    _label: string;
-    _values = {};
+    @State() _valid = true;
+    _dotTouched = false;
+    _dotPristine = true;
 
     componentWillLoad() {
         this._options = getDotOptionsFromFieldValue(this.options);
-        this._label = `dotCheckbox_${generateId()}`;
     }
 
-    // Todo: find how to set proper TYPE in TS
-    setValue(event): void {
-        const checkBoxVal = event.target.value.toString();
-        // Format values to be emmitted, this might change when implemented on the form
-        this._values = { ...this._values, [checkBoxVal]: !this._values[checkBoxVal] };
-        this.onChange.emit({ value: this._values });
+    hostData() {
+        return {
+            class: {
+                'dot-valid': this.isValid(),
+                'dot-invalid': !this.isValid(),
+                'dot-pristine': this._dotPristine,
+                'dot-dirty': !this._dotPristine,
+                'dot-touched': this._dotTouched,
+                'dot-untouched': !this._dotTouched
+            }
+        };
+    }
+
+    /**
+     * Reset properties of the field, clear value and emit events.
+     *
+     * @memberof DotSelectComponent
+     */
+    @Method()
+    reset(): void {
+        this._dotPristine = true;
+        this._dotTouched = false;
+        this._valid = true;
+        this.value = '';
+        this.emitValueChange();
+        this.emitStatusChange();
     }
 
     render() {
         return (
             <Fragment>
-                <label htmlFor={this._label}>{this.label}</label>
+                <label htmlFor={this.name}>{this.label}</label>
                 {this._options.map((item: DotOption) => {
-                    this._values = { ...this._values, [item.value]: this.value === item.value ? true : false };
+                    const trimmedValue = item.value.trim();
                     return (
-                        <div class='dot-checkbox__container'>
+                        <div class={this.getClassName()}>
                             <input
                                 type='checkbox'
-                                name={item.value}
-                                checked={this.value === item.value ? true : null}
+                                disabled={this.shouldBeDisabled()}
+                                id={this.value}
+                                checked={this.value.indexOf(trimmedValue) >= 0 ? true : null}
```

Comment:
`this.value.indexOf(trimmedValue) >= 0 ? true : null` can be `this.value.indexOf(trimmedValue) >= 0 || null`